src/index.spec.ts

```diff
@@ -0,0 +1,168 @@
+import 'mocha';
+import { assert } from 'chai';
+import { App, ExpressReceiver } from './index';
+import serverless from 'serverless-http';
+import {
+  delay,
+  createFakeMessageEvent,
+  createEventRequest,
+  importAppWithMockSlackClient,
+} from "./test-helpers";
+
+// How I'd like to write apps using koa-style middleware
+// @ts-ignore - This and the rest are changing signatures for consistency w/ koa, this wont run
+function exampleToContainScope() {  // tslint:disable:typedef
+  const app = new App();
+
+  // @ts-ignore
+  app.use(async (ctx, next) => {
+    const start = Date.now();
+    await next();
+    const end = Date.now();
+
+    console.log(`Request took ${end - start}ms`);
+  });
+
+  // @ts-ignore
+  app.message(
+      'Oh Hai!',
+      // @ts-ignore
+      async (ctx, next) => {
+        const start = Date.now();
+        await next();
+        const end = Date.now();
+
+        console.log(`Message request took ${end - start}ms`);
+      },
```

Comment:
you use this exact function twice, might be more readable to break it out and name it.

Reply:
I don't. I think inlining it makes it more clear. The intent is to show async middleware doing something going down the chain, then back up. 

In a real app, it would be different middleware, this is just a simple example of _something_.